src/rosdiscover/models/spawn_model.py

```diff
@@ -8,11 +11,50 @@
 from ..interpreter import model
 
 
+def _load_urdf_xml_from_parameter(c, parameter_name: str) -> t.Optional[ET]:
+    logger.debug(f'spawning model using parameter [{parameter_name}]')
+    urdf_contents = c.read(parameter_name).strip()
+    logger.debug(f'parsing URDF model from parameter [{parameter_name}]:'
+                 f'\n{urdf_contents}')
+
+    #  Workaround to find the <robot> begin tag
+    begin_tag = "<robot>"
+    begin_tag_starts_at = urdf_contents.find(begin_tag)
+    if begin_tag_starts_at == -1:
+        begin_tag_starts_at = urdf_contents.find("<robot ")
+    if begin_tag_starts_at != -1:
+        urdf_contents = urdf_contents[begin_tag_starts_at:]
+
+    # #94 workaround to deal with inclusion of warning when xacro.py is used
+    end_tag = '</robot>'
+    end_tag_starts_at = urdf_contents.rfind(end_tag)
+    end_tag_ends_at = end_tag_starts_at + len(end_tag)
+    urdf_contents = urdf_contents[:end_tag_ends_at]
+    logger.debug("New contents are:")
+    logger.debug(urdf_contents)
+    return ET.fromstring(urdf_contents)
+
+
+def _load_urdf_xml_from_file(c, filename: str) -> t.Optional[ET]:
+    if not os.path.isabs(filename):
+        logger.error(f"unable to load URDF XML from file [{filename}]: expected absolute path")
+        return None
+
+    logger.debug(f"loading URDF XML from file [{filename}]")
+    urdf_contents = c.read_file(filename)
+    logger.debug(f"loaded URDF XML contents from file [{filename}]:\n{urdf_contents}")
+    logger.debug("parsing URDF XML contents")
+    xml = ET.fromstring(urdf_contents)
+    logger.debug("parsed UDF XML contents")
+    return xml
```

Comment:
This might have the same problem as the other read, i.e., the urdf isn't surrounded by <robot>,

Reply:
From looking at one particular file, it seems to be well-formed, but yeah, I suppose that the same issue could occur. I'll tweak the code.